pkg/codegen/go/gen_program_expressions.go

```diff
@@ -350,18 +392,17 @@ func (g *generator) GenScopeTraversalExpression(w io.Writer, expr *model.ScopeTr
 
 func (g *generator) genScopeTraversalExpression(w io.Writer, expr *model.ScopeTraversalExpression, isInput bool) {
 	rootName := expr.RootName
-	// TODO splat
-	// if _, ok := expr.Parts[0].(*model.SplatVariable); ok {
-	// 	rootName = "__item"
-	// }
+
+	if _, ok := expr.Parts[0].(*model.SplatVariable); ok {
+		rootName = "val0"
+	}
 
 	genIDCall := false
 
-	var objType *schema.ObjectType
 	if resource, ok := expr.Parts[0].(*hcl2.Resource); ok {
 		isInput = false
 		if schemaType, ok := hcl2.GetSchemaForType(resource.InputType); ok {
-			objType, _ = schemaType.(*schema.ObjectType)
+			_, _ = schemaType.(*schema.ObjectType)
```

Comment:
Does this do anything useful?

Reply:
Ah yes, it can be removed. I missed this in the refactor of `g.genRelativeTraversal`.